### PR TITLE
Refactor msg matches

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Hi.

I found that your codebase is not very DRY. This is a first step, **it is a draft because I can do a lot more I guess**.

Would you like me to go on, or are these kind of changes not wanted?

---

Also: If these changes are wanted, I don't know how much progress there is in this repository. If there is a lot, I will chunk up my refactoring PRs so that we do not run into too many merge conflicts!

### CODE REVIEW

File: src/msg/cli.rs [hunk 1]
```diff
@@ -172,215 +172,371 @@ pub fn msg_subcmds<'a>() -> Vec<clap::App<'a, 'a>> {
...
+        ("list", Some(matches)) => msg_matches_list(app, matches),
+        ("move", Some(matches)) => msg_matches_move(app, matches),
+        ("read", Some(matches)) => msg_matches_read(app, matches),
+        ("reply", Some(matches)) => msg_matches_reply(app, matches),
+        ("save", Some(matches)) => msg_matches_save(app, matches),
+        ("search", Some(matches)) => msg_matches_search(app, matches),
+        ("send", Some(matches)) => msg_matches_send(app, matches),
+        ("template", Some(matches)) => msg_matches_template(app, matches),
+        ("write", Some(matches)) => msg_matches_write(app, matches),
+        (_other, _) => {
```
Comment (soywod) [inline src/msg/cli.rs:189]: This could be linked to `msg_matches_list`

**Comment by soywod** [comment]
I really like the way you refactor. It looks cleaner! I think it is a good idea to split your work in chunks to avoid conflict. But this part is more or less stable, it should not be a problem. Thanks a lot :heart:

**Comment by matthiasbeyer** [comment]
Awesome. I'll clean up for your comment and then submit the updated work for review.

**Comment by matthiasbeyer** [comment]
Okay, please have a close look at 007fd4d whether I messed up or not,... otherwise I'll mark this as ready for review and I'll be off the grid until tomorrow morning (CET).

**Comment by soywod** [comment]
I just finish a refactor from my side and I merge yours!

**Comment by soywod** [comment]
Thanks :tada:

**Pull request merged**